### ISSUE TITLE
rgblink infinite loop/memory leak

### ISSUE THREAD

**Issue by endrift** [rednex/rgbds#422]
Attempting to link with the attached file (after being gunzip'd) will cause rgblink to perpetually allocate memory, eat CPU, and seemingly never exit.
[c.o.gz](https://github.com/rednex/rgbds/files/3593131/c.o.gz)

**Comment by ISSOtm** [comment]
The error is an unterminated string, which RGBLINK does not check for, and since EOF didn't yield a \0, the loop never exited.

**Comment by ISSOtm** [comment]
```
issotm@sheik-kitty tmp.HUkUe3eik0% rgblink -V
rgblink v0.3.9-26-ge1a2fde
issotm@sheik-kitty tmp.HUkUe3eik0% rgblink c.o
error: c.o: Cannot read section name: Unexpected end of file
```
Fixed by f2e1b7d8683b46f9631b5f550605f396fcf21f65
